cloudsmith/resource_license_policy.go

```diff
@@ -0,0 +1,249 @@
+
+	org := requiredString(d, Organization)
+
+	req := pc.APIClient.OrgsApi.OrgsLicensePolicyDelete(pc.Auth, org, d.Id())
+	_, err := pc.APIClient.OrgsApi.OrgsLicensePolicyDeleteExecute(req)
+	if err != nil {
+		return err
+	}
+
+	checkerFunc := func() error {
+		req := pc.APIClient.OrgsApi.OrgsLicensePolicyRead(pc.Auth, org, d.Id())
+		if _, resp, err := pc.APIClient.OrgsApi.OrgsLicensePolicyReadExecute(req); err != nil {
+			if is404(resp) {
+				return nil
+			}
+			return err
+		}
+		return errKeepWaiting
+	}
+	if err := waiter(checkerFunc, defaultDeletionTimeout, defaultDeletionInterval); err != nil {
+		return fmt.Errorf("error waiting for license policy (%s) to be deleted: %w", d.Id(), err)
+	}
+
+	return nil
+}
+
+func resourceLicensePolicyRead(d *schema.ResourceData, m interface{}) error {
+	pc := m.(*providerConfig)
+
+	org := requiredString(d, Organization)
+
+	req := pc.APIClient.OrgsApi.OrgsLicensePolicyRead(pc.Auth, org, d.Id())
+
+	licensePolicy, resp, err := pc.APIClient.OrgsApi.OrgsLicensePolicyReadExecute(req)
+	if err != nil {
+		if is404(resp) {
+			d.SetId("")
+			return nil
+		}
+
+		return err
+	}
+
+	_ = d.Set(CreatedAt, licensePolicy.GetCreatedAt().String())
+	_ = d.Set(Description, licensePolicy.GetDescription())
+	_ = d.Set(Name, licensePolicy.GetName())
+	_ = d.Set(OnViolationQuarantine, licensePolicy.GetOnViolationQuarantine())
+	_ = d.Set(AllowUnknownLicenses, licensePolicy.GetAllowUnknownLicenses())
+	_ = d.Set(SlugPerm, licensePolicy.GetSlugPerm())
+	_ = d.Set(SpdxIdentifiers, flattenStrings(licensePolicy.GetSpdxIdentifiers()))
+	_ = d.Set(UpdatedAt, licensePolicy.GetUpdatedAt().String())
+
+	// organization is not returned from the read
+	// endpoint, so we can use the values stored in resource state. We rely on
+	// ForceNew to ensure if either changes a new resource is created.
+	_ = d.Set(Organization, org)
+
+	return nil
+}
+
+//nolint:funlen
+func resourceLicensePolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceLicensePolicyCreate,
+		Read:   resourceLicensePolicyRead,
+		Update: resourceLicensePolicyUpdate,
+		Delete: resourceLicensePolicyDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: importLicensePolicy,
+		},
+
+		Schema: map[string]*schema.Schema{
+			CreatedAt: {
+				Type:        schema.TypeString,
+				Description: "The time the policy was created at.",
+				Computed:    true,
+			},
+			Description: {
+				Type:         schema.TypeString,
+				Description:  "The description of the license policy.",
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			Name: {
+				Type:         schema.TypeString,
+				Description:  "The name of the license policy.",
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			OnViolationQuarantine: {
+				Type:        schema.TypeBool,
+				Description: "On violation of the license policy, quarantine violating packages.",
+				Optional:    true,
+				Default:     true,
```

Comment:
do you actually want to set a client side default here or set these fields to computed and let the server side default pass through?

Reply:
general point for all fields, not just this one... NAB though, just raising the question

Reply:
Probably best letting serverside handle defaults, in case they get changed around. Will change this now thanks!